worker/pkg/workflows/datasync/activities/run-sql-init-table-stmts/init-statement-builder.go

```diff
@@ -199,6 +200,37 @@ func (b *initStatementBuilder) RunSqlInitTableStatements(
 					return nil, fmt.Errorf("unable to exec ordered truncate statements: %w", err)
 				}
 			}
+			if !sqlopts.InitSchema && (sqlopts.TruncateBeforeInsert || sqlopts.TruncateCascade) {
```

Comment:
why are we only doing this if we are not also initializing the schema? The schema init is idempotent so what if the schema already exists? maybe I'm just missing some context here.